### DISCUSSION TITLE
请问下umi4 request如何配置取消请求呢！

### DISCUSSION THREAD

**Discussion by codedart2018** [umijs/umi#9982] [Q&A]
我看说umi4 request 已经换成了axios .
我在request配置里加上
```
  cancelToken
```
只要一加就报错。整个项目就挂掉了！
有时候一个页面有多个请求，如果某个请求遇到权限不足被后台拦截了，我想通过返回的懒觉终止后面的请求。需要如何处理呢！

**Comment by fz6m** [comment]
响应拦截器，如果没有权限，就把某个全局标识置为 `false` 。

请求拦截器，每次请求前都检查这个全局标识，如果是 `false` 就不进行请求并提示。

https://umijs.org/docs/max/request#requestinterceptors

**Reply by codedart2018** [reply]
> 响应拦截器，如果没有权限，就把某个全局标识置为 `false` 。
> 
> 请求拦截器，每次请求前都检查这个全局标识，如果是 `false` 就不进行请求并提示。
> 
> https://umijs.org/docs/max/request#requestinterceptors

这个拦截会终止后面的请求嘛？

**Comment by zyk2891** [comment]
如果想终止前面的响应应该怎么办？

**Reply by zyk2891** [reply]
就比如有个接口服务器响应比较慢 我点击了另外一个查询接口之后导致前面较慢响应的接口覆盖了后面的 怎么终止之前的这个响应

**Reply by fz6m** [reply]
1. 现代数据流更推荐底层自己封装 axios / fetch ，顶层使用 [react-query](https://umijs.org/docs/max/react-query) 来管理请求状态数据（不推荐使用 umi request ），使用 [jotai](https://github.com/pmndrs/jotai) / [valtio](https://umijs.org/docs/max/valtio) 来管理全局数据。

2. 在底层的 axios / fetch 中，可以传递 cancel token 取消上一个请求。
